cyclops-ui/src/components/k8s-resources/common/PodTable/PodLogs.tsx

```diff
@@ -99,31 +114,32 @@ const PodLogs = ({ pod }: PodLogsProps) => {
   };
 
   const onLogsTabsChange = (container: string) => {
-    axios
-      .get(
-        "/api/resources/pods/" +
-          logsModal.namespace +
-          "/" +
-          logsModal.pod +
-          "/" +
-          container +
-          "/logs",
-      )
-      .then((res) => {
-        if (res.data) {
-          let log = "";
-          res.data.forEach((s: string) => {
-            log += s;
-            log += "\n";
+    setLogsSignalController((prevController) => {
+      if (prevController) {
+        prevController.abort();
+      }
+
+      const controller = new AbortController();
+      return controller;
+    });
```

Comment:
Thanks for highlighting this issue. It might arise in some cases, so to completely avoid the situation, we can add the following code (which I used in the file `PodLogs.tsx`, lines 164–180):

```typescript
const controller = new AbortController();
setLogsSignalController(() => controller);

logStream(
  pod.name,
  pod.namespace,
  pod.containers[0].name,
  (log) => {
    setLogs((prevLogs) => [...prevLogs, log]);
  },
  (err) => {
    setError(mapResponseError(err));
  },
  controller,
);
```

We handle the required modification by aborting the previous signal controller, and then creating and passing a **new signal controller** from the closure. This ensures that the new `AbortController` is always passed to the `logStream` function, making sure an active abort signal controller is present in the `logStream`.

Reply:
Sure, feel free to add it to the PR